daml/Marketplace/Custody/Model.daml

```diff
@@ -113,9 +114,13 @@ template LifecycleRequest
               election : Claim Observation Date Id -> Claim Observation Date Id -> Claim Observation Date Id
               election _ _ = deserialize choice -- will throw if the claims don't match
           Lifecycle.Result{remaining, pending} <- Lifecycle.lifecycle spot election (deserialize description.claims) t
-          (_, postAssetVersion) <- exercise versionsCid . AssetDescription.LookupOrInsert $ serialize remaining
-          payoutCids <- forA pending \(qty, asset) -> do
-            payoutDepositCid :: _ <- exercise safekeepingDepositCid $ AssetDeposit_Split [qty * assetDeposit.asset.quantity]
+          (_, postAssetVersion, _) <- exercise versionsCid AssetDescription.LookupOrInsert with newClaims = serialize remaining, assetDescription = description
+          let payoutSplitQuantities = (\(qty, _) -> qty * assetDeposit.asset.quantity) <$> pending
+          -- Next step could be:
+          -- create SettlementInstruction with buyer = customer; seller = issuer; ..
+          splitResultDeposits <- exercise safekeepingDepositCid AssetDeposit_Split with quantities = payoutSplitQuantities
```

Comment:
So if I understand correctly, you're letting `AssetDeposit_Split` handle the splitting by passing it a list instead of calling it individually each time, correct?

Reply:
Exactly